OBSDOCS-93: Update install docs version and formatting (#67879)
openshift/openshift-docs: modules/cluster-logging-deploy-cli.adoc

:_mod-docs-content-type: PROCEDURE
[id="cluster-logging-deploy-cli_{context}"]
= Installing the {logging} using the CLI

> You can use the {product-title} CLI to install the {es-op} and the {clo}.

Reviewer: ```suggestion
You can use the {oc-first} to install the {es-op} and the {clo}.
```
Since we use `oc` after the first mention, I'd use the attribute here.

metric, which would cause conflicts.
<2> String. You must specify this label as shown to ensure that cluster monitoring scrapes the `openshift-operators-redhat` namespace.

. Apply the `Namespace` object by running the following command:
+
[source,terminal]
----

> $ oc apply -f <filename>

Reviewer: Wouldn't hurt to footnote this user value with what to substitute with here. I would likely do this for each command in each section. Perhaps you can make a generic note? "Insert the filename of the object you just created." or something like that?

Author: This seems to be pretty hit or miss throughout the docs wrt. whether there's a standard here. There are some instances where people seem to add footnotes, but plenty where they're not added. I would err on the side of I think we can assume that level of knowledge, and this is what has been done for any other doc set I've worked on previously, so I'd prefer not to start adding footnotes for this unless we end up adding a rule in the doc guidelines or somewhere about it.

    openshift.io/cluster-monitoring: "true"
----

. Apply the `Namespace` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>
----

. Install the {es-op} by creating the following objects:

.. Create an `OperatorGroup` object for the {es-op}:

> +

Reviewer: I think "Example `OperatorGroup` object" is missing here before the YAML.

kind: Subscription
metadata:
  name: elasticsearch-operator
  namespace: openshift-operators-redhat <1>
spec:
  channel: <channel> <2>
  installPlanApproval: Automatic <3>
  source: redhat-operators <4>
  sourceNamespace: openshift-marketplace
  name: elasticsearch-operator
----
<1> You must specify the `openshift-operators-redhat` namespace.
<2> Specify `stable`, or `stable-y.z` as the channel, where `y` is the major version and `z` is the minor version. See the following note.
<3> `Automatic` allows the Operator Lifecycle Manager (OLM) to automatically update the Operator when a new version is available. `Manual` requires a user with appropriate credentials to approve the Operator update.
<4> Specify `redhat-operators`. If your {product-title} cluster is installed on a restricted network, also known as a disconnected cluster,
specify the name of the CatalogSource object created when you configured the Operator Lifecycle Manager (OLM).
+
[NOTE]
====
Specifying `stable` installs the current version of the latest stable release. Using `stable` with `installPlanApproval: "Automatic"` automatically upgrades your Operators to the latest stable major and minor release.

> Specifying `stable-y.z` installs the current minor version of a specific major release. Using `stable-y.z` with `installPlanApproval: "Automatic"` automatically upgrades your Operators to the latest stable minor release within the major `y` release.

Reviewer: Do you mean the minor or patch/z release? We usually refer to minor as y, and z as "patch" or just "z" releases. It's major-minor-patch: 4.14.3, for example.

Author: For logging we only refer to major and minor, per https://docs.openshift.com/container-platform/4.14/logging/cluster-logging-upgrading.html
Do you know if the major-minor-patch thing is documented anywhere? It might be helpful to point to as a follow up.

====

.. Apply the `Subscription` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>
----
+
The {es-op} is installed to the `openshift-operators-redhat` namespace and copied to each project in the cluster.

. Install the {clo} by creating the following objects:

.. Create an `OperatorGroup` object for the {clo}:

> +

Reviewer: I think "Example `OperatorGroup` object" is missing here before the YAML.

<1> You must specify the `openshift-logging` namespace.
<2> Specify `stable`, or `stable-5.<x>` as the channel.
<3> Specify `redhat-operators`. If your {product-title} cluster is installed on a restricted network, also known as a disconnected cluster, specify the name of the CatalogSource object you created when you configured the Operator Lifecycle Manager (OLM).

.. Apply the `Subscription` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>
----
+
The {clo} is installed to the `openshift-logging` namespace.

. Create a `ClusterLogging` custom resource (CR):
+
[NOTE]
====

> This default `ClusterLogging` CR configuration should support a wide array of environments. Review the topics on tuning and

Reviewer: ```suggestion
This default `ClusterLogging` CR configuration should support a wide array of environments. Review the topics on tuning and configuring {logging} components for information about modifications you can make to the {logging}.
```
Perhaps all on one line while you're here? It looks nice in the preview, but the line break seems odd.

> configuring {logging} components for information about modifications you can make to the {logging}.

Reviewer: ```suggestion

```

====

> +

Reviewer: "Example `ClusterLogging` YAML" (or "custom resource") is missing

    elasticsearch:
      nodeCount: 3 <5>
      storage:
        storageClassName: <storage_class_name> <6>
        size: 200G
      resources: <7>
        limits:
          memory: 16Gi
        requests:
          memory: 16Gi
      proxy: <8>
        resources:
          limits:
            memory: 256Mi
          requests:
             memory: 256Mi
      redundancyPolicy: SingleRedundancy
  visualization:
    type: kibana  <9>
    kibana:
      replicas: 1
  collection:
    logs:
      type: fluentd  <10>
      fluentd: {}
----
<1> The name must be `instance`.

> <2> The {logging} management state. In some cases, if you change the {logging} defaults, you must set this to `Unmanaged`.

Reviewer: here, too, the line break is working for the preview, but it could probably be deleted

The number of primary shards for the index templates is equal to the number of Elasticsearch data nodes.
====

. Apply the `ClusterLogging` custom resource (CR) by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>
----
+
This creates the {logging} components, the `Elasticsearch` CR and components, and the Kibana interface.

.Verification

. Verify the {es-op} installation:
+
[source,terminal]
----
$ oc get csv --all-namespaces
----
+
.Example output
[source,terminal]
----

> NAMESPACE                                               NAME                                            DISPLAY                  VERSION               REPLACES   PHASE

Reviewer: ```suggestion
NAMESPACE                                               NAME                                            DISPLAY                  	VERSION               		REPLACES   	   PHASE
```
These titles aren't aligned once you get to Version if you scroll over. Looks like another tab or two is needed.

openshift-apiserver-operator                            elasticsearch-operator.5.5.0-202007012112.p0    OpenShift Elasticsearch Operator   5.5.0-202007012112.p0               Succeeded
openshift-apiserver                                     elasticsearch-operator.5.5.0-202007012112.p0    OpenShift Elasticsearch Operator   5.5.0-202007012112.p0               Succeeded
openshift-authentication-operator                       elasticsearch-operator.5.5.0-202007012112.p0    OpenShift Elasticsearch Operator   5.5.0-202007012112.p0               Succeeded
openshift-authentication                                elasticsearch-operator.5.5.0-202007012112.p0    OpenShift Elasticsearch Operator   5.5.0-202007012112.p0               Succeeded
...
----
+
There should be an {es-op} instance in each namespace. The version number might be different than shown.

. Verify the {clo} installation.
+
There should be a Red Hat OpenShift Logging Operator in the `openshift-logging` namespace. The Version number might be different than shown.
+
[source,terminal]
----
$ oc get csv -n openshift-logging
----
+
.Example output
[source,terminal]
----

> NAMESPACE                                               NAME                                         DISPLAY                  VERSION               REPLACES   PHASE

Reviewer: ```suggestion
NAMESPACE                            NAME                                         DISPLAY                  VERSION               REPLACES   PHASE
```
This one, too. A lot of space between namespace and name, but once we get to Version, things are not aligned.

...
openshift-logging                                       clusterlogging.5.1.0-202007012112.p0         OpenShift Logging          5.1.0-202007012112.p0              Succeeded
...
----

. Verify the installation by listing the pods in the *openshift-logging* project. Run the following command:
+
[source,terminal]
----
$ oc get pods -n openshift-logging
----
+
You should see several pods for components of the {logging}, similar to the following list:
+

> .Example output

Reviewer: Out of scope of this PR, but the output is also not aligned once you get to the collector-587vb row (and following rows).
